### DISCUSSION TITLE
Total/Progress Callback's reverse behavior between compression and decompression operations

### DISCUSSION THREAD

**Discussion by Polar-Pumpkin** [rikyoz/bit7z#258] [Q&A]
I noticed that during the decompression operation, the Progress Callback is first called to return the total size, and then the Total Callback is called multiple times to return the decompression progress. Is this expected and designed? (bit7z v4.0.8 msvc2022_x64 from GitHub Release)

**Comment by rikyoz** [comment]
Hi, sorry for the late reply!
Um, that's weird and definitely not the expected behaviour.
I just did a quick test and could not replicate the problem.
What archive format are you extracting from?

**Reply by Polar-Pumpkin** [reply]
Sorry for the late reply. The formats I use include 7z and zip. I just wrote a piece of code to test the callback function of the compression and decompression process. I found that it is actually working properly. I am sorry for the trouble and misleading information.

I reported this problem because the progress bar in the interface I wrote in C# did not work properly. By adding log information to the callback function (on the C# side), I found that the logs output by the `Total` and `Progress` callback functions had opposite phenomena. I forgot to test it further and reported the problem here. I am really sorry for taking up your time and causing you trouble. 🙇

After making the above judgment, I swapped the operations of the two callback functions on my progress bar in the C# code, so my progress bar has always been in a state of not working properly. Now after reorganizing this piece of code, my progress bar behaves normally, which means that there is actually no problem. Thank you very much for taking the time to answer my question!

**Reply by rikyoz** [reply]
No problem at all! 
Thank you for using bit7z! 🙏
